spring-integration-ftp/src/test/java/org/springframework/integration/ftp/dsl/FtpTests.java

```diff
@@ -192,6 +192,36 @@ public void testFtpOutboundFlow() {
 		registration.destroy();
 	}
 
+	@Test
+	public void testFtpOutboundFlowWithChmod() {
+		IntegrationFlow flow = f -> f
+				.handle(Ftp.outboundAdapter(sessionFactory(), FileExistsMode.FAIL)
+						.useTemporaryFileName(false)
+						.fileNameExpression("headers['" + FileHeaders.FILENAME + "']")
+						.chmod(644)
```

Comment:
Yes, but is is done in the client with Integer.toOctalString(), which should return  1204

```
@Override
	protected void doChmod(RemoteFileTemplate<FTPFile> remoteFileTemplate, final String path, final int chmod) {
		remoteFileTemplate.executeWithClient((ClientCallbackWithoutResult<FTPClient>) client -> {
			String chModCommand = "chmod " + Integer.toOctalString(chmod) + " " + path;
			try {
				client.sendSiteCommand(chModCommand);
			}
			catch (IOException e) {
				throw new UncheckedIOException("Failed to execute '" + chModCommand  + "'", e);
			}
		});
	}
```

Reply:
Not sure why you said that it must return `1204`, because this is what I have in my test:
```
Integer.toOctalString(0644)
...
Result: 644
```

Reply:
I just made some more tests, then I'm right, give a look to my debug session

![image](https://user-images.githubusercontent.com/2885322/66931270-261f5a80-f036-11e9-92ca-6e67440bc4ae.png)

Reply:
> > looks like the permission is not being set properly
> 
> Maybe you expect permissions which are not really a result of the provided octal?

I just made a test on Linux, chmod 0644 test.file, and the result was this
-rw-r--r--   1 root root     0 Oct 16 17:30 test.file

so that translated to the code should make sense

Reply:
Shouldn't that be `chmod 644 test.file` on Linux instead?
That is exactly what we get with the `"chmod " + Integer.toOctalString(chmod) + " " + path;`

The point is that such a number must be *octal* and only the way to restrict it on Java side is to provide a value with a leading `0`

Reply:
> Shouldn't that be `chmod 644 test.file` on Linux instead?
> That is exactly what we get with the `"chmod " + Integer.toOctalString(chmod) + " " + path;`
> 
> The point is that such a number must be _octal_ and only the way to restrict it on Java side is to provide a value with a leading `0`

wether i do it with linux (0)644, the result is always the same (-rw-r--r--), that's the weird thing, is maybe the fake internal client ? I'll try with a local filezilla thing

Reply:
looks like it's the implementation, 
I'll update you when I'm done with filezilla
https://mina.apache.org/ftpserver-project/ftpserver_site_commands.html

Reply:
Ah! You mean an embedded MINA FTP Server we use for testing doesn't implement `chmod` site command properly?

Reply:
I think so, even though I would expect an exception that's it's not implement, or maybe it just bypasses it?
I'll let you know by the way after the test

Reply:
Yes, 
I can confirm, the problem is the built-in server, we'll have to skip the test or what other options do we have?
![image](https://user-images.githubusercontent.com/2885322/66987472-27e23000-f0c1-11e9-8267-0bb1da88c83d.png)

Reply:
Well, we can just make the test against mocks. 
See sample in the `FtpServerOutboundTests.testInt3088MPutNotRecursive()`.

It really not so important if command has reached the server or not. Our goal is to ensure that your DSL fix brings for us a proper chmod command in the end.

Reply:
@artembilan  turning back to this point, so is enough as it is with the .chmod on the flow ? the mockito should be sending the site command right ?

![image](https://user-images.githubusercontent.com/2885322/67290687-c6ff9100-f4e0-11e9-84ec-396383d54b64.png)

Reply:
yes, that was what I tried, I commited, let me know!